File: src/iml/Database/DataProviderSQLite.java
/*
 *  Created by Somesh Chatterjee.
 */
package iml.Database;

import iml.Database.TableStrategies.CompanyMasterStrategy;
import iml.Database.TableStrategies.ITableStrategy;
import iml.Database.TableStrategies.IncomingTableStrategy;
import iml.Database.TableStrategies.MOCQualityStrategy;
import iml.Database.TableStrategies.OutgoingTableStrategy;
import iml.Database.TableStrategies.Parser;
import iml.Database.TableStrategies.TransporterStrategy;

/**
 *
 * @author somesh
 */
public class DataProviderSQLite
{
    private final TableManager _tableManager;
    private final ITableStrategy _incomingTableStrategy;
    private final ITableStrategy _outgoingTableStrategy;
    private final ITableStrategy _companyMasterTableStrategy;
    private final ITableStrategy _transporterTableStrategy;
    private final ITableStrategy _mocQualityTableStrategy;
    private final Parser _parser;

    public DataProviderSQLite()
    {
        _incomingTableStrategy = new IncomingTableStrategy();
        _outgoingTableStrategy = new OutgoingTableStrategy();
        _companyMasterTableStrategy = new CompanyMasterStrategy();
        _transporterTableStrategy = new TransporterStrategy();
        _mocQualityTableStrategy = new MOCQualityStrategy();
        _parser = new Parser();
        _tableManager = new TableManager();
        _tableManager.CreateTableIfNotExist(_incomingTableStrategy);
        _tableManager.CreateTableIfNotExist(_outgoingTableStrategy);
        _tableManager.CreateTableIfNotExist(_companyMasterTableStrategy);
        _tableManager.CreateTableIfNotExist(_transporterTableStrategy);
        _tableManager.CreateTableIfNotExist(_mocQualityTableStrategy);
    }

    public String GetIncomingData()
    {
        Object[][] result = _tableManager.ReadRecords(_incomingTableStrategy);
        return _parser.ConvertToString(result);
    }
    
    public String GetMOCAndQualityCode()
    {
        Object[][] result = _tableManager.ReadRecords(_mocQualityTableStrategy);
        return _parser.ConvertToString(result);
    }
    
    public String GetOutgoingData()
    {
        Object[][] result = _tableManager.ReadRecords(_outgoingTableStrategy);
        return _parser.ConvertToString(result);
    }
    
    public String GetCompanyMaster()
    {
        Object[][] result = _tableManager.ReadRecords(_companyMasterTableStrategy);
        return _parser.ConvertToString(result);
    }

    public String GetTransporterDetails()
    {
        Object[][] result = _tableManager.ReadRecords(_transporterTableStrategy);
        return _parser.ConvertToString(result);
    }

    public void SetIncomingData(String values)
    {
        Object[] valuesInJavaArray = _incomingTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.CreateRecord(_incomingTableStrategy, valuesInJavaArray);
    }
    
    public void SetOutgoingData(String values)
    {
        Object[] valuesInJavaArray = _outgoingTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.CreateRecord(_outgoingTableStrategy, valuesInJavaArray);
    }
    
    public void SetCompanyMasterData(String values)
    {
        Object[] valuesInJavaArray = _companyMasterTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.CreateRecord(_companyMasterTableStrategy, valuesInJavaArray);
    }
    
    public void SetMOCAndQualityCodeData(String values)
    {
        Object[] valuesInJavaArray = _mocQualityTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.CreateRecord(_mocQualityTableStrategy, valuesInJavaArray);
    }
    
    public void SetTransporterValuesInTable(String values)
    {
        Object[] valuesInJavaArray = _transporterTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.CreateRecord(_transporterTableStrategy, valuesInJavaArray);
    }
    
    public void UpdateIncomingData(int id, String values)
    {
        Object[] valuesInJavaArray = _incomingTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.UpdateRecord(_incomingTableStrategy, id, valuesInJavaArray);
    }
    
    public void UpdateOutgoingData(int id, String values)
    {
        Object[] valuesInJavaArray = _outgoingTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.UpdateRecord(_outgoingTableStrategy, id, valuesInJavaArray);
    }
    
    public void UpdateCompanyMasterData(int id, String values)
    {
        Object[] valuesInJavaArray = _companyMasterTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.UpdateRecord(_companyMasterTableStrategy, id, valuesInJavaArray);
    }
    
    public void UpdateMOCAndQualityCodeData(int id, String values)
    {
        Object[] valuesInJavaArray = _mocQualityTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.UpdateRecord(_mocQualityTableStrategy, id, valuesInJavaArray);
    }
    
    public void UpdateTransporterData(int id, String values)
    {
        Object[] valuesInJavaArray = _transporterTableStrategy.GetObjectArrayFromJSONArray(_parser.ConvertToArray(values));
        _tableManager.UpdateRecord(_transporterTableStrategy, id, valuesInJavaArray);
    }    
}

File: src/HTML/DataProviderSQLite.js
/* 
 *  Created by Somesh Chatterjee.
 */


var DataProviderSQLite = (function ()
{    
    var _dataProvider = parent.DataProviderSQLite;
    
    function GetIncomingData()
    {
        var result = _dataProvider.GetIncomingData();       
        return DataParser.ConvertToArray(result);        
    }
    
    function GetMOCAndQualityCode()
    {
        var result = _dataProvider.GetMOCAndQualityCode();       
        return DataParser.ConvertToArray(result);        
    }
    
    function GetOutgoingData()
    {
        var result = _dataProvider.GetOutgoingData();       
        return DataParser.ConvertToArray(result);              
    }    
    
    function GetCompanyMaster()
    {
        var result = _dataProvider.GetCompanyMaster();       
        return DataParser.ConvertToArray(result);
    }
    
    function GetTransporterDetails()
    {
        var result = _dataProvider.GetTransporterDetails();       
        return DataParser.ConvertToArray(result);
    }
    
    function SetIncomingData(newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.SetIncomingData(result);
    }
    
    function SetOutgoingData(newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.SetOutgoingData(result);
    }
    
    function SetCompanyMasterData(newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.SetCompanyMasterData(result);
    }
    
    function SetMOCAndQualityCodeData(newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.SetMOCAndQualityCodeData(result);
    }
    
    function SetTransporterValuesInTable(newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.SetTransporterValuesInTable(result);
    }
    
    function UpdateIncomingData(id, newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.UpdateIncomingData(id, result);
    }
    
    function UpdateOutgoingData(id, newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.UpdateOutgoingData(id, result);
    }
    
    function UpdateCompanyMasterData(id, newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.UpdateCompanyMasterData(id, result);
    }
    
    function UpdateMOCAndQualityCodeData(id, newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.UpdateMOCAndQualityCodeData(id, result);
    }
    
    function UpdateTransporterData(id, newRow)
    {
        var result = DataParser.ConvertToString(newRow);
        _dataProvider.UpdateTransporterData(id, result);
    }
    
    return {
        GetIncomingData : GetIncomingData,
        GetCompanyMaster: GetCompanyMaster,
        GetTransporterDetails: GetTransporterDetails,
        GetOutgoingData: GetOutgoingData,
        GetMOCAndQualityCode: GetMOCAndQualityCode,
        SetIncomingData: SetIncomingData,
        SetOutgoingData: SetOutgoingData,
        SetCompanyMasterData: SetCompanyMasterData,
        SetMOCAndQualityCodeData: SetMOCAndQualityCodeData,
        SetTransporterValuesInTable: SetTransporterValuesInTable,
        UpdateIncomingData: UpdateIncomingData,
        UpdateOutgoingData: UpdateOutgoingData,
        UpdateCompanyMasterData: UpdateCompanyMasterData,
        UpdateMOCAndQualityCodeData: UpdateMOCAndQualityCodeData,
        UpdateTransporterData: UpdateTransporterData
    };
    
})();